src/middlewared/middlewared/plugins/smb_/registry_share.py

```diff
@@ -343,6 +346,10 @@ async def share_to_smbconf(self, conf_in, globalconf=None):
 
         if osc.IS_FREEBSD:
             data['vfsobjects'] = ['aio_fbsd']
+        elif is_clustered:
+            conf["glusterfs: volume"] = data["cluster_volume"]
```

Comment:
`KeyError` because of minor typo here:
```suggestion
            conf["glusterfs: volume"] = data["cluster_volname"]
```